static/app/components/profiling/flamegraph/flamegraphOverlays/FlamegraphWarnings.tsx

```diff
@@ -13,6 +14,19 @@ interface FlamegraphWarningProps {
 export function FlamegraphWarnings(props: FlamegraphWarningProps) {
   const orgSlug = useOrganization().slug;
   const params = useParams();
+  const profiles = useProfiles();
+
+  if (profiles.type === 'loading') {
+    return null;
+  }
+
+  if (profiles.type === 'errored') {
+    return (
+      <Overlay>
+        <p>{profiles.error || t('Failed to fetch profile')}</p>
```

Comment:
Nit: fetch seems like it implies the backend failed but it could be the import as well

```suggestion
        <p>{profiles.error || t('Failed to load profile')}</p>
```